syntax = "proto3";

message ServerStatus {
  float speed = 1;
  float cadence = 2;
  int32 target_gear = 3;
  int32 current_gear = 4;

  enum State {
    BIKING = 0;
    STOPPED = 1;
    ASLEEP = 2;
  }

  State state = 5;

  enum ButtonState {
    RELEASED = 0;
    PRESSED = 1;
  }

  ButtonState up_shift_button = 6;
  ButtonState down_shift_button = 7; 
}